myoFlex/myoFlex.ino

```diff
@@ -1,69 +1,124 @@
+ 
+// THIS PART ONLY RUNS ONCE
+void setup() {
+Serial.begin(9600);     //starts up serial communication between arduino and computer
+  Serial.println("linking servos");
+  thumb.attach(thumbPin, min, max);    //defines where the servos are and their max/mins
+  pointer.attach(pointerPin, min, max);
+  middle.attach(middlePin, min, max);
+  ring.attach(ringPin, min, max);
+  pinky.attach(pinkyPin, min, max);
+  Serial.println("Starting");       //makes the hand relax all of the way
+  handPosition(max,max,max,max,max);
+  trigger = setTrigger();
+}
+ 
+// THIS PART OF THE CODE LOOPS CONTINIOUSLY
+void loop() {
+  Serial.println(analogRead(myoIn));
+  if(check(currentVoltage = analogRead(myoIn))){
+    state++;
+    state = state%4;
+    Serial.print("state is ");
+    Serial.println(state);
+    //Serial.println(currentVoltage);        //prints in the serial moniter
+    if(state == 0){
+      Serial.println("Open");
+      handPosition(max,max,max,max,max);
+    }
+    else if(state == 1){
+      Serial.println("Point");
+      handPosition(max,min,max,max,max);
+    }
+    else if(state == 2){
+      Serial.println("Pinch");
+      handPosition(min,min,min,max,max);
+    }
+    else if(state == 3){
+      Serial.println("Fist");
+      handPosition(min,min,min,min,min);
+    }
+    else {
+      state = 1; //edgecase if logic fail
+      handPosition(max,max,max,max,max);
+    }
+  }
+}
+int setTrigger(){
+  float average = 0;
+  int i, j;
+  do{
+    average = 0;
+    for(j = 1; j <= 3; j++){
+      Serial.print("Reading impulse ");
+      Serial.print(j);
+      Serial.println(" in the next 3 seconds");
+      for(i = 3; i != 0; i--){
+        Serial.println(i);
+        delay(1000);
+      }
+      Serial.println("Reading");
+      average = average + analogRead(myoIn);
+      Serial.println(average/3);
+    }
+  } while((average/3) < 200);
+  return (int)(average/3);
+}
+
+int check(int voltage){
+  if(voltage > trigger){
+    return 1;
+  }
+  return 0;
+}
+
+void handPosition(int thumbPos,int pointerPos,int middlePos,int ringPos,int pinkyPos){
+  //function hand position takes in 5 integer inputs for each finger and moves the servos according to
+  //each input. for the servos, maximum extension is 1000 and minimum (all the way in) is 2000.
+  thumb.writeMicroseconds(thumbPos);   //this is a function from the servo.h library basically gives the pulse width
+  pointer.writeMicroseconds(pointerPos);
+  middle.writeMicroseconds(middlePos);
+  ring.writeMicroseconds(ringPos);
+  pinky.writeMicroseconds(pinkyPos);
+  Serial.println("changing");
+  delay(1000);       //this is so the servos have time to get to their set position
+  thumb.writeMicroseconds(thumb.readMicroseconds());
+  pointer.writeMicroseconds(pointer.readMicroseconds());
+  middle.writeMicroseconds(middle.readMicroseconds());
+  ring.writeMicroseconds(ring.readMicroseconds());
+  pinky.writeMicroseconds(pinky.readMicroseconds());
+  delay(250);
+}
```

Comment:
Correct me if i am wrong: I think this lets the servos try and get to where they are set and then sets them wherever they get to? 

GREAT!

Reply:
yep

Reply:
we can even do a subtraction operation on our passed values if we want the hand to let go a bit further i.e 

pinky.writeMicroseconds(pinky.readMicroseconds() - 10);